oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -614,9 +640,12 @@ impl Runtime {
         &self,
         node_id: NodeId,
         readers: &[Handle],
-    ) -> Result<Vec<ChannelReadStatus>, OakStatus> {
-        self.validate_handles_access(node_id, readers)?;
-        self.validate_can_read_from_channels(node_id, readers)?;
+    ) -> Result<Vec<ChannelReadStatus>, ErroneousChannelStatus> {
+        self.validate_handles_access(node_id, readers)
+            .map_err(|e| ErroneousChannelStatus::Err(e))?;
+        self.validate_can_read_from_channels(node_id, readers)
+            .map_err(|e| ErroneousChannelStatus::Err(e))?;
```

Comment:
(Pre-existing) This label check has the same problem I was concerned about elsewhere &ndash; if there's a single channel in the collection being waited on that cannot be read from due to invalid labels, then the called will get a failure code and have no immediate way of figuring out which channel was restricted.

Reply:
@tiziano88 @daviddrysdale WDYT?

Reply:
I guess that means we should also extend [ChannelReadStatus](https://github.com/project-oak/oak/blob/a1bd9e0cc24cd5da62191bda5dae719dd291f9ad/oak/proto/oak_abi.proto#L51) to account for permission denied, right?